### PR TITLE
INT-6416 Adding config options for liveness and readiness probes

### PR DESCRIPTION
**Pull request by hectorhuol**
Adding config options for liveness and readiness probes:
* I have updated the `deployment.yaml` template to add the new configuration options
* I have updated the `values.yaml` with the default values for the new configuration options
* I have updated the `README.md` file to add the new configuration options

JIRA: https://issues.sonatype.org/browse/INT-6416
Build: https://jenkins.ci.sonatype.dev/job/integrations/job/cloud/job/Helm3%20Charts/job/Feature%20Snapshot%20Builds/job/INT-6416-adding-liveness-readiness-options/

### CODE REVIEW

**Review by adrianpowell — APPROVED** [review]

File: charts/nexus-iq/values.yaml [hunk 1]
```diff
@@ -28,6 +28,20 @@ iq:
...
+    failureThreshold: 3
+    timeoutSeconds: 2
+    successThreshold: 1
+  # Configures the liveness probe for IQ pod
+  readinessProbe:
+    initialDelaySeconds: 10
+    periodSeconds: 10
+    failureThreshold: 3
+    timeoutSeconds: 2
+    successThreshold: 1
```
Comment (hajimeo) [inline charts/nexus-iq/values.yaml:44]: Does above mean those values will be the default if users do not specify?
If so, aren't above a bit too strict for larger IQ + H2?

Comment (hectorhuol) [reply charts/nexus-iq/values.yaml:44]: Hi @hajimeo, yes those will be the default values for users that do not specify them on a custom values file, and on this initial PR, I am just using the default values that Kubernetes have for those params, so we do not change the current behavior. Look at this: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes

Regarding if these values are too strict, maybe you are right, and we have here an opportunity to have more accurate values. I am going to check with the team in detail. 

Also, if you have any suggestions for those numbers, please let us know.

Comment (hectorhuol) [reply charts/nexus-iq/values.yaml:44]: Edit: I was wrong when I said I was using the default values, actually I updated the values for `initialDelaySeconds` and `initialDelaySeconds`, default values were `initialDelaySeconds:0` and `timeoutSeconds:1`

File: charts/nexus-iq/values.yaml [hunk 2]
```diff
@@ -28,6 +28,20 @@ iq:
...
     - name: JAVA_OPTS
       value: "-Djava.util.prefs.userRoot=${SONATYPE_WORK}/javaprefs"
+  # Configures the liveness probe for IQ pod
+  livenessProbe:
+    initialDelaySeconds: 10
+    periodSeconds: 10
+    failureThreshold: 3
+    timeoutSeconds: 2
+    successThreshold: 1
+  # Configures the liveness probe for IQ pod
```
Comment (eduard-tita) [inline charts/nexus-iq/values.yaml:38]:
```suggestion
  # Configures the readiness probe for IQ pod
```

**Review by eduard-tita — APPROVED** [review]
LGTM with one minor comment

File: charts/nexus-iq/values.yaml (hunk 1 above)
Comment (jflinchbaugh) [reply charts/nexus-iq/values.yaml:44]: @hajimeo do you have ideas for more appropriate values to specify here? we need to balance that time needed for the first migration startup vs daily operation. In daily operation, the server failing for many minutes before being restarted could be too long, but I guess that's "liveness" not "readiness".
To facilitate the long migration time, we _may_ want to drastically extend readiness times and thresholds. Is there a downside to waiting too long for the startup? probably not.

File: charts/nexus-iq/templates/deployment.yaml [hunk 3]
```diff
@@ -47,10 +47,20 @@ spec:
             httpGet:
               path: /ping
               port: admin
+            initialDelaySeconds: {{ .Values.iq.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.iq.livenessProbe.periodSeconds }}
+            failureThreshold: {{ .Values.iq.livenessProbe.failureThreshold }}
+            timeoutSeconds: {{ .Values.iq.livenessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.iq.livenessProbe.successThreshold }}
```
[the remark this answers is not in the document]
Comment (hectorhuol) [reply charts/nexus-iq/templates/deployment.yaml:54]: Sure I am going to check and make the change. This means that on `values.yaml` we will need to comment the lines related to `livenessProbe` and `readinessProbe`, right?

File: charts/nexus-iq/values.yaml (hunk 1 above)
Comment (hectorhuol) [reply charts/nexus-iq/values.yaml:44]: I just updated the `readinessProbe` default values to try to cover a larger IQ setup. I think with those values we should cover a lot of scenarios. In any case, if needed, now users can override those values with a custom values file

File: charts/nexus-iq/templates/deployment.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (hectorhuol) [reply charts/nexus-iq/templates/deployment.yaml]: I just updated the code. Now if we do not have the values for `livenessProbe` or `readinessProbe`, the POD will start with the default values

File: charts/nexus-iq/templates/deployment.yaml [hunk 4]
```diff
@@ -49,10 +49,16 @@ spec:
             httpGet:
               path: /ping
               port: admin
+    {{- if .Values.iq.livenessProbe }}
+            {{- toYaml .Values.iq.livenessProbe | nindent 12 }}
+    {{- end }}
```
Comment (jflinchbaugh) [inline charts/nexus-iq/templates/deployment.yaml:54]: nit: i'd advocate to indent the templating bits to be in-line with their surroundings (further right).

Comment (hectorhuol) [reply charts/nexus-iq/templates/deployment.yaml]: Ok, makes sense, done 😄

**Review by jflinchbaugh — APPROVED** [review]
+1 approve even more now!

**Pull request merged**